initfiles/componentfiles/thor/run_thor

```diff
@@ -69,41 +69,59 @@ while [ 1 ]; do
     nohup $instancedir/thormaster_$THORNAME MASTER=$THORMASTER:$THORMASTERPORT 2> /dev/null 1>/dev/null &
 
     thorpid=$!
-    if [ "$thorpid" -ne "0" ]; then 
+    if [[ "$thorpid" -ne "0" ]]; then
         echo thormaster$LCR process started pid = $thorpid
         
         echo $thorpid > $PID_NAME
         echo $thorpid > $PID_NAME_BASE
         wait $thorpid
         errcode=$?
-        if [ "$errcode" -eq "0" ]; then  # allows for CTRL-C or thorstop
-            echo thor exited cleanly - script stopping
+        case $errcode in
+        # TEC_Clean
+        0)  echo "Thormaster ($thorpid) exited cleanly"
+            rm -rf $instancedir/uslaves.start $PID_NAME > /dev/null 2>&1
             exit 0
-        fi
-        echo master exited with errorcode = $errcode
-        if [ ! -e $SENTINEL ]; then
-            echo $SENTINEL 'has been removed (1) - script stopping'
+            ;;
+        # TEC_CtrlC
+        1)  echo "Thormaster ($thorpid) Interrupted, Ctrl-C caught"
             $deploydir/stop_thor $deploydir
-            exit 0
-        fi
-
-        echo 'stopping thor(slaves) for restart'
-        $deploydir/stop_thor $deploydir keep_sentinel
-
-        if [ 0 != $autoSwapNode ]; then
-            echo "Running autoswap $THORNAME"
-            compname=`basename $PWD`
-            swapnode auto $DALISERVER $compname
-            errcode=$?
-            if [ 0 != ${errcode} ]; then
-                echo auto swap node failed, errcode=${errcode}
-            fi
-        fi
+            exit 1
+            ;;
+        # TEC_Idle, TEC_Watchdog, TEC_Swap, TEC_DaliDown
+        2|3|5|6)    [[ $errcode -lt 4 ]] && echo "Thormaster ($thorpid) Idle or lost connection to slave(s)"
+                    [[ $errcode -eq 5 ]] && echo "Thormaster ($thorpid) Exited with TEC_Swap"
```

Comment:
Trivial: only message that mentions internal enum code, prob. best if says ".. Swap node required"